### ISSUE TITLE
Ability to set all the HTTPS options myself

### ISSUE THREAD

**Issue by binarykitchen** [AveVlad/gulp-connect#172]
In https://nodejs.org/api/https.html#https_https_createserver_options_requestlistener you can see that the `https.createServer()` call supports may options. I.E. `rejectUnauthorized`. Unfortunately your code does not allow this. Can we change this line

```
      @server = (http2 || https).createServer
        key: @https.key || fs.readFileSync __dirname + '/certs/server.key'
        cert: @https.cert || fs.readFileSync __dirname + '/certs/server.crt'
        ca: @https.ca || fs.readFileSync __dirname + '/certs/ca.crt'
        passphrase: @https.passphrase || 'gulp'
        @app
```

to

```
      @server = (http2 || https).createServer(@https, @app)
```

instead? That would make things a lot easier for us to configure. Also because I generate my local certs without a passphrase. The default one `gulp` is breaking it. Then mine are self-signed, and thus do not require the ca option.

**Comment by justinmchase** [comment]
Yeah, you're right. Setting the default pass like that is a bug.

What you're asking for is a breaking change but probably a good one overall. I would accept the PR but the `@https` object should probably have the default `key,cert,ca,passphrase` set if the key is not set still. But not all individually like that.

Fire up a PR and I'll merge it!

**Comment by binarykitchen** [comment]
alright, here you go https://github.com/AveVlad/gulp-connect/pull/178

**Comment by justinmchase** [comment]
Published in  [v3.2.2](https://github.com/AveVlad/gulp-connect/releases/tag/v3.2.2)
